addon/-private/model.ts

```diff
@@ -20,16 +20,11 @@ interface HasManyContract {
 }
 
 export default class Model extends EmberObject {
-  identity: RecordIdentity;
+  identity: RecordIdentity = { id: '', type: '' };
 
   private _store?: Store;
   private _relatedRecords: Dict<HasManyContract> = {};
 
-  constructor(settings: ModelSettings) {
```

Comment:
Instead of the constructor seems that we should use `init`. But since in this case the name of the property in the object with which the model is going to be initialized (in an `assign`-like manner) matches the name of the property in the object, we don't even need an initializer.

We only have to assign an initial identity that is going to be immediately replaced, but that makes typescript happy.

Reply:
Good catch. This does seem like the "happy path" for both Ember and TS. Thanks @cibernox!

Reply:
I am bothered a bit by the default, bogus `identity` value.

Reply:
If we don't add that, we'd have to make identity optional with `identity?: RecordIdentity`, and add presence checks in several places.

Reply:
I am pretty sure we are testing recent Ember versions. Why our tests are not caching this?

Reply:
@dgeb I found a way to not initialize the identity to a bogus value. 
`identity!: RecordIdentity` tells TS that it can be sure that identity will be present even if not explicitly assigned in the constructor.

Reply:
> `identity!: RecordIdentity` tells TS that it can be sure that identity will be present even if not explicitly assigned in the constructor.

@cibernox ah, that's a nice trick!

> I am pretty sure we are testing recent Ember versions. Why our tests are not caching this?

@tchak We haven't been using the dummy app in the tests, but we should probably do so to fully exercise DI in a "real" app.